tasks/system_probe.py

```diff
@@ -698,9 +701,43 @@ def build_security_ebpf_files(ctx, build_dir, parallel_build=True):
     build_security_offset_guesser_ebpf_files(ctx, build_dir)
 
 
-def build_object_files(ctx, parallel_build):
+def setup_clang(ctx, target):
+    # check if correct version is already present
+    res = ctx.run("/opt/datadog-agent/embedded/bin/clang-bpf --version", warn=True)
+    if res.ok:
+        version_str = res.stdout.split("\n")[0].split(" ")[2].strip()
+        if version_str == target:
+            return
+
+    arch = arch_mapping.get(platform.machine())
+    if arch == "x64":
+        arch = "amd64"
+
+    clang_present = ctx.run(f"stat /opt/datadog-agent/embedded/bin/clang-{target}", warn=True)
+    llc_present = ctx.run(f"stat /opt/datadog-agent/embedded/bin/llc-{target}", warn=True)
+
+    if clang_present.failed or llc_present.failed:
+        # download correct version from dd-agent-omnibus S3 bucket
+        clang_url = f"https://dd-agent-omnibus.s3.amazonaws.com/llvm/clang-{target}.{arch}"
+        ctx.sudo(f"wget -q {clang_url} -O /opt/datadog-agent/embedded/bin/clang-{target}")
+        ctx.sudo(f"chmod 0755 /opt/datadog-agent/embedded/bin/clang-{target}")
+
+        llc_url = f"https://dd-agent-omnibus.s3.amazonaws.com/llvm/llc-{target}.{arch}"
+        ctx.sudo(f"wget -q {llc_url} -O /opt/datadog-agent/embedded/bin/llc-{target}")
+        ctx.sudo(f"chmod 0755 /opt/datadog-agent/embedded/bin/llc-{target}")
+
+    # make symlinks to the target binary
+    ctx.sudo("rm /opt/datadog-agent/embedded/bin/clang-bpf", warn=True)
```

Comment:
I'm not sure we want to support multiple clang/llc version, so we don't need to keep symlink strategy here, just overwrite the file clang-bpf and we doesn't to complexity, so we don't need to pass bpf_clang_version parameter